packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart

```diff
@@ -830,29 +836,35 @@ class _StoryOfYourFlexWidgetState extends State<StoryOfYourFlexWidget>
     final verticalAxisDescription = Align(
       alignment: Alignment.bottomLeft,
       child: Container(
-        height: maxHeight - mainAxisArrowIndicatorSize,
+        margin: const EdgeInsets.only(top: mainAxisArrowIndicatorSize + margin),
         width: crossAxisArrowIndicatorSize,
         child: Column(
           children: [
             Expanded(
               child: ArrowWrapper.unidirectional(
                 arrowColor: verticalColor,
-                child: RotatedBox(
-                  quarterTurns: 3,
-                  child: Text(
-                    properties.verticalDirectionDescription,
-                    overflow: TextOverflow.ellipsis,
-                    textAlign: TextAlign.center,
-                    textScaleFactor: largeTextScaleFactor,
-                    style: TextStyle(
-                      color: verticalTextColor,
+                child: Flexible(
+                  flex: maxHeight <= minHeightToAllowTruncating ? 1 : 0,
+                  child: RotatedBox(
+                    quarterTurns: 3,
+                    child: Text(
+                      properties.verticalDirectionDescription,
+                      overflow: TextOverflow.ellipsis,
+                      textAlign: TextAlign.center,
+                      textScaleFactor: largeTextScaleFactor,
+                      style: TextStyle(
+                        color: verticalTextColor,
+                      ),
                     ),
                   ),
                 ),
                 type: ArrowType.down,
               ),
             ),
-            _buildAxisAlignmentDropdown(Axis.vertical),
+            Flexible(
+              flex: maxHeight <= minHeightToAllowTruncating ? 1 : 0,
```

Comment:
this expression is repeated a few times.  Maybe add a widget that is
Truncateable({bool truncate, Widget child})
as a bonus I can imagine adding some fancier logic for how we truncate later on.

Reply:
+1